File: linkedList/3.py
from typing import List


def countStudents(students: List[int], sandwiches: List[int]) -> int:
    number_no_match = 0
    index_sandwich = 0

    while True:
        if students[0] == sandwiches[index_sandwich]:
            index_sandwich += 1
            students.pop(0)
            number_no_match = 0
        else:
            number_no_match += 1
            temp = students.pop(0)
            students.append(temp)

        if number_no_match == len(students) or len(students) == 0:
            break

    return len(students)


print(countStudents([1, 1, 1, 0, 0, 1], [1, 0, 0, 0, 1, 1]))

File: linkedList/3.cpp
#include <bits/stdc++.h>

using namespace std;

int countStudents(vector<int>& students, vector<int>& sandwiches) {
    int cont = 0;
    int numberNoMatch = 0;

    list<int> bocaditos(sandwiches.begin(), sandwiches.end());
    list<int> estudiantes(students.begin(), students.end());

    while(true) {
        if(bocaditos.front() == estudiantes.front()) {
            bocaditos.pop_front();
            estudiantes.pop_front();
            numberNoMatch = 0;
        } else {
            int temp = estudiantes.front();
            estudiantes.pop_front();
            estudiantes.push_back(temp);

            numberNoMatch++;
        }

        if(numberNoMatch == estudiantes.size() || estudiantes.empty())
            break;

    }


    return estudiantes.size();
}

int main() {
    ios_base::sync_with_stdio(false);


    vector<int> students = {1,1,0,0};
    vector<int> sandwiches = {0,1,0,1};
    
    printf("%d", countStudents(students, sandwiches));


    return 0;
}